ikarus/finiteelements/mechanics/materials/neohooke.hh

```diff
@@ -96,7 +101,12 @@ struct NeoHookeT : public Material<NeoHookeT<ST>>
     static_assert(Concepts::EigenMatrixOrVoigtNotation3<Derived>);
     if constexpr (!voigt) {
       if constexpr (!Concepts::EigenVector<Derived>) {
-        const auto logdetF = log(sqrt(C.determinant()));
+        const auto detC = C.determinant();
+        if (Dune::FloatCmp::lt(static_cast<double>(detC), 0.0, 1e-10))
```

Comment:
Since you are repeating that functionality I would move it to a private member function and call it check positive determinant of C.